### PR TITLE
REF use centralised addRadio function more

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This makes use of the CRM_Core_Form::addRadio function more

Before
----------------------------------------
Various classes creating radio and group elements themselves

After
----------------------------------------
using the centralised function to create Group of radio elements

ping @demeritcowboy @eileenmcnaughton

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Test Result (18 failures / ±0)CRM_Contribute_Form_Contribution_MainTest.testExpiredPriceSetCRM_Event_Form_ParticipantTest.testSubmitWithPayment with data set #0CRM_Event_Form_ParticipantTest.testSubmitWithPayment with data set #1CRM_Event_Form_ParticipantTest.testSubmitWithFailedPayment with data set #0CRM_Event_Form_ParticipantTest.testSubmitWithFailedPayment with data set #1CRM_Event_Form_ParticipantTest.testParticipantOfflineReceipt with data set #0CRM_Event_Form_ParticipantTest.testParticipantOfflineReceipt with data set #1CRM_Event_Form_ParticipantTest.testSubmitPartialPayment with data set #0CRM_Event_Form_ParticipantTest.testSubmitPartialPayment with data set #1CRM_Event_Form_ParticipantTest.testSubmitPendingPartiallyPaidAddPayment with data set #0CRM_Event_Form_ParticipantTest.testSubmitPendingPartiallyPaidAddPayment with data set #1CRM_Event_Form_ParticipantTest.testSubmitPendingAddPayment with data set #0CRM_Event_Form_ParticipantTest.testSubmitPendingAddPayment with data set #1CRM_Event_Form_ParticipantTest.testSubmitUnpaidPriceChangeWhileStillPendingCRM_Event_Form_ParticipantTest.testPaymentAllocationOnMultiLineItemEventCRM_Event_Form_ParticipantTest.testSubmitWithDeferredRecognitionCRM_Event_Form_ParticipantTest.testTransferParticipantRegistrationCRM_Financial_Form_PaymentFormsTest.testEventPaymentForms
--

File: CRM/Contribute/Form/ManagePremiums.php [hunk 1]
```diff
@@ -86,13 +86,17 @@ public function buildQuickForm() {
...
+    $image['image'] = ts('Upload from my computer');
+    $imageJS['image'] = ['onclick' => 'add_upload_file_block(\'image\');', 'class' => 'required'];
+    $image['thumbnail'] = ts('Display image and thumbnail from these locations on the web:');
+    $imageJS['thumbnail'] = ['onclick' => 'add_upload_file_block(\'thumbnail\');', 'class' => 'required'];
+    $image['default_image'] = ts('Use default image');
+    $imageJS['default_image'] = ['onclick' => 'add_upload_file_block(\'default\');', 'class' => 'required'];
+    $image['noImage'] = ts('Do not display an image');
+    $imageJS['noImage'] = ['onclick' => 'add_upload_file_block(\'noImage\');', 'class' => 'required'];
+
+    $this->addRadio('imageOption', ts('Premium Image'), $image, [], NULL, FALSE, FALSE, $imageJS);
```
Comment (demeritcowboy) [inline CRM/Contribute/Form/ManagePremiums.php:99]: There's one too many false's here. Clicking on the radios doesn't run any javascript.

(Normally two wrongs make a right, but not this time.)

File: CRM/Event/Form/Registration/Register.php [hunk 2]
```diff
@@ -651,18 +651,14 @@ public static function buildAmount(&$form, $required = TRUE, $discountId = NULL)
...
-          );
+          $elements[$fee['amount_id']] = CRM_Utils_Money::format($fee['value']) . ' ' . $fee['label'];
+          $elementJS[$fee['amount_id']] = $totalAmountJs;
         }
       }
       $form->assign('eventFeeBlockValues', json_encode($eventFeeBlockValues));
 
       $form->_defaults['amount'] = $form->_values['event']['default_fee_id'] ?? NULL;
-      $element = &$form->addGroup($elements, 'amount', ts('Event Fee(s)'), '<br />');
+      $element = &$form->addRadio('amount', ts('Event Fee(s)'), $elements, [], '<br />', FALSE, $elementJS);
```
Comment (demeritcowboy) [inline CRM/Event/Form/Registration/Register.php:661]: Just eyeballing looks ok, but I can't figure out how to get here. Everything always gets treated as a price set even when not using price sets.

File: CRM/Contribute/Form/ManagePremiums.php (hunk 1 above)
Comment (seamuslee001) [reply CRM/Contribute/Form/ManagePremiums.php]: Thanks @demeritcowboy fixed

File: CRM/Contribute/Form/ContributionPage/Amount.php [hunk 3]
```diff
@@ -70,10 +70,10 @@ public function buildQuickForm() {
       $this->addRule("value[$i]", ts('Please enter a valid money value (e.g. %1).', [1 => CRM_Utils_Money::formatLocaleNumericRoundedForDefaultCurrency('99.99')]), 'money');
 
       // default
-      $default[] = $this->createElement('radio', NULL, NULL, NULL, $i);
+      $default[$i] = NULL;
     }
 
-    $this->addGroup($default, 'default');
+    $this->addGroup('default', '', $default);
```
Comment (demeritcowboy) [inline CRM/Contribute/Form/ContributionPage/Amount.php:76]: This is causing an error when creating a new contribution page. Is there a reason the params are reversed?

File: CRM/Event/Form/ManageEvent/Fee.php [hunk 4]
```diff
@@ -292,10 +292,10 @@ public function buildQuickForm() {
       $this->addRule("value[$i]", ts('Please enter a valid money value for this field (e.g. %1).', [1 => CRM_Utils_Money::formatLocaleNumericRoundedForDefaultCurrency('99.99')]), 'money');
 
       // default
-      $default[] = $this->createElement('radio', NULL, NULL, NULL, $i);
+      $default[$i] = NULL;
     }
 
-    $this->addGroup($default, 'default');
+    $this->addRadio('default', '', $default);
```
Comment (demeritcowboy) [inline CRM/Event/Form/ManageEvent/Fee.php:299]: I didn't notice before but probably same here.

Comment (demeritcowboy) [reply CRM/Event/Form/ManageEvent/Fee.php:299]: Oh no never mind not here.

File: CRM/Contribute/Form/ContributionPage/Amount.php (hunk 3 above)
Comment (seamuslee001) [reply CRM/Contribute/Form/ContributionPage/Amount.php:76]: This is due to wrong function being called should be addRadio will fix shortly

File: CRM/Member/Form/MembershipBlock.php [hunk 5]
```diff
@@ -160,8 +160,7 @@ public function buildQuickForm() {
         $this->assign('auto_renew', $this->_renewOption);
       }
       $this->addGroup($membership, 'membership_type', ts('Membership Types'));
-      $this->addGroup($membershipDefault, 'membership_type_default', ts('Membership Types Default'))
-        ->setAttribute('allowClear', TRUE);
+      $this->addGroup('membership_type_default', ts('Membership Types Default'), $membershipDefault, ['allowClear' => TRUE]);
```
Comment (demeritcowboy) [inline CRM/Member/Form/MembershipBlock.php:163]: Here too. Should be addRadio.

Calling it a day so no hurry.

Comment (seamuslee001) [reply CRM/Member/Form/MembershipBlock.php]: should be sorted now

File: CRM/Contribute/Form/ContributionPage/Amount.php (hunk 3 above)
Comment (seamuslee001) [reply CRM/Contribute/Form/ContributionPage/Amount.php]: should be sorted now

**Comment by demeritcowboy** [comment]
I didn't test every possible combination but otherwise looks good.

I still don't know how to reach that event one but I suppose if you can't reach it then it's not a blocker.

**Comment by seamuslee001** [comment]
@eileenmcnaughton thoughts on the above comment?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 so I'm a bit nervous about how many forms are changed & the potential for something weird not to have been r-run here.

On the event register - I think it's desirable that that code not be hit and if it isn't we should confirm & fully remove (the treat things the same logic is the intended logic).

Perhaps we can make a list / table of places affected & identify which ones have been tested (it's fine if its you not a separate reviewer)

File: CRM/Contribute/Form/ContributionBase.php [hunk 6]
```diff
@@ -881,9 +881,9 @@ public function buildComponentForm($id, $form) {
 
         // radio button for Honor Type
         foreach ($form->_values['soft_credit_types'] as $value) {
-          $honorTypes[$value] = $form->createElement('radio', NULL, NULL, $softCreditTypes[$value], $value);
+          $honorTypes[$value] = $softCreditTypes[$value];
         }
-        $form->addGroup($honorTypes, 'soft_credit_type_id', NULL)->setAttribute('allowClear', TRUE);
+        $form->addRadio('soft_credit_type_id', NULL, $honorTypes, ['allowClear' => TRUE]);
```
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/ContributionBase.php:886]: I have tested this change

**Comment by demeritcowboy** [comment]
I tested all of them except of the event one, but many of them have several paths to get there with multiple possibilities. The ones where it was clear there's only one path and seemed straightforward were:

CRM/Contact/Form/Task/AddToParentClass.php
CRM/Contribute/Form/ManagePremiums.php
CRM/Custom/Form/Field.php

**Comment by eileenmcnaughton** [comment]
OK - that seems pretty thorough to me then - merging - we should make it easier for later people to see the event one is not in use

**Pull request merged**